samples/dashboards/demo/personal-portfolio/demo.js

```diff
@@ -0,0 +1,579 @@
+    id: stock.ISIN,
+    idType: 'ISIN',
+    ...(weight && { weight })
+}));
+
+
+const kpiValueFormatter = t => '€' + Highcharts.numberFormat(t, 2, '.', ',');
+
+
+// Return the sum of the last indices of arrays
+const getCurrentTotal = arrOfArr => {
+    let sum = 0;
+    arrOfArr.forEach(arr => {
+        sum += arr.at(-1);
+    });
+
+    return sum;
+};
+
+(async () => {
+    // eslint-disable-next-line no-undef
+    const timeSeriesConnector = new HighchartsConnectors.Morningstar
+        .TimeSeriesConnector({
+            ...commonMSOptions,
+            series: {
+                type: 'Price'
+            },
+            securities: getHoldings(),
+            currencyId: 'EUR',
+            startDate: '2022-01-01',
+            endDate: '2023-12-31'
+        });
+
+    await timeSeriesConnector.load();
+
+
+    const { Date: dates, ...companies } =
+        timeSeriesConnector.table.getColumns();
+
+    const processedData = Object.fromEntries(
+        Object.entries(companies).map(([key, values]) => [
+            key,
+            values.map((value, i) => [dates[i], value])
+        ])
+    );
+
+    const holdings = [],
+        investedAmounts = [],
+        dataGridData = [];
+
+    stockCollection.forEach(stock => {
+        const { holding, investedAmount } = generatePortfolio(
+            basicInvestmentPlan,
+            processedData[stock.SecID]
+        );
+
+        holdings.push(holding);
+        investedAmounts.push(investedAmount);
+
+    });
+
+    const investedAmountTotal = getCurrentTotal(investedAmounts),
+        lastHoldingTotal = getCurrentTotal(holdings),
+        annualInvestment = 200 * 12 * holdings.length;
+
+    // Generate columns for the datagrid
+    stockCollection.forEach((stock, i) => {
+        const len = holdings[i].length,
+            lastHolding = holdings[i][len - 1],
+            ISIN = stock.ISIN,
+            tradingSymbol = stock.tradingSymbol;
+
+        dataGridData.push([
+            tradingSymbol,
+            ISIN,
+            Math.round(lastHolding / lastHoldingTotal * 100)
+        ]);
+    });
+
+    // Generate portfolio for risk score
+    const portfolio = {
+        name: 'PersonalPortfolio',
+        currency: 'EUR',
+        totalValue: lastHoldingTotal,
+        // Holdings with equal weights
+        holdings: getHoldings(100 / stockCollection.length)
+    };
+
+    // Creating chart and KPI options
+    const walletChartOptions = {
+        plotOptions: {
+            series: {
+                marker: {
+                    enabled: false
+                }
+            }
+        },
+        chart: {
+            height: 400
+        },
+        accessibility: {
+            description: 'displaying portfolio performance and investments'
+        },
+        rangeSelector: {
+            animate: false,
+            x: 0,
+            y: 0,
+            inputEnabled: false,
+            dropdown: 'never',
```

Comment:
Also probably redundant, because the chart will never get so narrow that it goes to dropdown mode. And if it did, it would be for a good reason.
```suggestion
```